lib/Haskell/Prim/Foldable.agda

```diff
@@ -19,54 +19,54 @@ record Foldable (t : Set → Set) : Set₁ where
   field
     foldMap : ⦃ Monoid b ⦄ → (a → b) → t a → b
 
-  foldr : (a → b → b) → b → t a → b
-  foldr f z t = foldMap ⦃ MonoidEndo ⦄ f t z
+open Foldable ⦃ ... ⦄ public
 
-  foldl : (b → a → b) → b → t a → b
-  foldl f z t = foldMap ⦃ MonoidEndoᵒᵖ ⦄ (flip f) t z
+{-# COMPILE AGDA2HS Foldable existing-class #-}
 
-  any : (a → Bool) → t a → Bool
-  any = foldMap ⦃ MonoidDisj ⦄
+foldr : ⦃ Foldable t ⦄ → (a → b → b) → b → t a → b
+foldr f z t = foldMap ⦃ it ⦄ ⦃ MonoidEndo ⦄ f t z
```

Comment:
I think this has to be resolved before making it to the release. You mention the way to go is 'default methods', but we only care about this when we have multiple minimal records, do you mean that we should apply the same treatment here to properly resolve this and put the definitions back into the record?

Reply:
Yeah I was thinking of applying the same method here. Even if there is only a single minimal record, having a version of the record where all of the methods are fields still makes sense since this allows the defaults to be overridden.

Reply:
I've refactored `lib/` to use proper default methods, it makes things a bit clunky to use (because you now need to use the record-as-module syntax to construct minimal instances) but it works. It would be nice to someday make the instance translation a bit more liberal to allow for more complicated expression that may streamline such default constructions (e.g. see `foldMap=_` in my commit).